src/main/scala/io/iohk/ethereum/domain/Blockchain.scala

```diff
@@ -260,13 +274,15 @@ class BlockchainImpl(
 
   override def save(hash: ByteString, evmCode: ByteString): Unit = evmCodeStorage.put(hash, evmCode)
 
-  override def saveBestBlockNumber(number: BigInt): Unit =
-    appStateStorage.putBestBlockNumber(number)
+  override def saveBestBlockNumber(number: BigInt): Unit = {
```

Comment:
Minor detail: Maybe it would be better rename this method to `saveBestKnownBlock` to distinguish it from `AppStateStorage.saveBestBlockNumber`?